hazelcast/test/src/HazelcastTests3.cpp

```diff
@@ -534,6 +542,38 @@ TEST_F(ClientReplicatedMapTest, testClientPortableWithoutRegisteringToNode)
     ASSERT_EQ(666, samplePortable->a);
 }
 
+TEST_F(ClientReplicatedMapTest, testDeregisterListener)
+{
+    auto map = client->get_replicated_map(get_test_name()).get();
+
+    ASSERT_FALSE(
+      map->remove_entry_listener(spi::ClientContext(*client).random_uuid())
+        .get());
+
+    boost::latch map_clearedLatch(1);
+
+    entry_listener listener;
+
+    listener.on_map_cleared([&map_clearedLatch](map_event&& event) {
+        ASSERT_EQ(get_test_name(), event.get_name());
+        ASSERT_EQ(entry_event::type::CLEAR_ALL, event.get_event_type());
+        const std::string& hostName =
+          event.get_member().get_address().get_host();
+        ASSERT_TRUE(hostName == "127.0.0.1" || hostName == "localhost");
+        ASSERT_EQ(5701, event.get_member().get_address().get_port());
+        ASSERT_EQ(1, event.get_number_of_entries_affected());
+        std::cout << "Map cleared event received:" << event << std::endl;
+        map_clearedLatch.count_down();
+    });
+
+    auto listenerRegistrationId =
+      map->add_entry_listener(std::move(listener), true).get();
+    map->put(1, 1).get();
+    map->clear().get();
+    ASSERT_OPEN_EVENTUALLY(map_clearedLatch);
+    ASSERT_TRUE(map->remove_entry_listener(listenerRegistrationId).get());
```

Comment:
Removal of listener is not verified. I thinks it needs to be refactored as following. Let's say a functor object which is attached to listener is executed when new entry is inserted into map. Then steps might be like below :

- Insert a new entry and verify that functor object is executed
- Remove listener and insert new entry
- Verify that functor is not executed.

Reply:
Fixed